lambdas/src/apis/images/controllers/images.ts

```diff
@@ -0,0 +1,119 @@
+import log4js from "log4js";
+import { Request, Response } from "express";
+import { SmartContentServerFetcher } from "../../../SmartContentServerFetcher";
+import { Environment, EnvironmentConfig } from "../../../Environment";
+import fs from "fs";
+import sharp from "sharp";
+import fetch from "node-fetch"
+
+const LOGGER = log4js.getLogger("ImagesController");
+
+const validSizes = ["128", "256", "512"];
+
+class ServiceError extends Error {
+  statusCode: number;
+
+  constructor(message: string, code: number = 400) {
+    super(message);
+    this.statusCode = code;
+  }
+}
+
+function validateSize(size: string) {
+  if (!validSizes.includes(size)) {
+    throw new ServiceError("Invalid size");
+  }
+}
+
+async function existPath(path: string): Promise<boolean> {
+  try {
+    await fs.promises.access(path, fs.constants.F_OK | fs.constants.W_OK);
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+async function ensureDirectoryExists(directory: string): Promise<void> {
+  const alreadyExist = await existPath(directory);
+  if (!alreadyExist) {
+    await fs.promises.mkdir(directory);
```

Comment:
On the content server we had a few problems because we would get many requests and we would end up calling `mkdir` many times. I would add a `try/catch` to avoid failing on those situations